src/Uno.UI/UI/Xaml/Media/ImageSource.crossruntime.cs

```diff
@@ -107,11 +107,8 @@ private void Open()
 				{
 					asyncImg.ContinueWith(t =>
 					{
-						if (t.IsCompletedSuccessfully)
-						{
-							OnOpened(t.Result);
-						}
-					}, ct);
+						OnOpened(t.IsFaulted ? ImageData.FromError(t.Exception) : t.Result);
+					}, ct, TaskContinuationOptions.None, TaskScheduler.FromCurrentSynchronizationContext());
```

Comment:
@ramezgerges This somehow makes 

![image](https://github.com/user-attachments/assets/6a30a4f8-24eb-4784-a086-10fca58b00ca)

fail to load.